### PR TITLE
Rotorsoft/issue5293 - Create a shell script to read the current database schema and generate a mermaid ERD in the wiki space

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #5293

## Description of Changes
- created sql query to read pg metadata and generate mermaid erd schema
- created shell script to save results in wiki/ERD.md

## Test Plan
- Execute yarn workspace commonwealth db-doc to generate a fresh copy of ./knowledge_base/wiki/ERD.md

## Other Considerations
- not integrated in the ci/cd pipeline yet, can be added to `db-all` script if necessary

### CODE REVIEW

File: packages/commonwealth/package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "create-migration": "npx sequelize migration:generate --name",
     "datadog-db-setup": "chmod u+x scripts/setup-datadog-postgres.sh && ./scripts/setup-datadog-postgres.sh",
     "db-all": "yarn reset-db && yarn load-db && yarn migrate-db",
+    "db-doc": "chmod u+x scripts/gen-mermaid-erd.sh && ./scripts/gen-mermaid-erd.sh > ../../knowledge_base/wiki/ERD.md",
```
Comment (gdjohnson) [inline packages/commonwealth/package.json:19]: If we could get a few lines of documentation for this script, that'd be awesome. See `Package-Scripts.md` in our knowledge base.

File: knowledge_base/wiki/ERD.md [hunk 2]
```diff
@@ -0,0 +1,503 @@
+# Autogenerated Mermaid ERD
```
Comment (gdjohnson) [inline knowledge_base/wiki/ERD.md:1]: I think a slightly. more descriptive file name would help contextualize what this document is and where it's headed. Maybe even just "Database ERD"?

File: packages/commonwealth/scripts/gen-mermaid-erd.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+SQL=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )/mermaid-erd.sql
+MERMAID=$(PGPASSWORD=edgeware psql -h localhost -d commonwealth -U commonwealth -t -A -f $SQL)
+echo -e "# Autogenerated Mermaid ERD\n\n$(date)\n\n\`\`\`mermaid\n$MERMAID\n\`\`\`"
```
Comment (gdjohnson) [inline packages/commonwealth/scripts/gen-mermaid-erd.sh:5]: Possible to add a "Last updated" before the timestamp, just to be super abundantly clear what the timestamp signifies?

**Review by gdjohnson — CHANGES_REQUESTED** [review]
This is awesome! Super cool idea elegantly implemented.

My only big-picture concern is that it's such an enormous ERD, it's difficult to view or navigate visually. Curious if you have any advice, or what you've been using to visually navigate the generated ERD.

**Comment by rbennettcw** [comment]
Nice.

Looks like the script assumes that Postgres is running directly on the host machine? For those of us who run Postgres in Docker (Compose) it would be nice to have a script (or documentation including the script) that can work in that context.

**Comment by Rotorsoft** [comment]
> Nice.
> 
> Looks like the script assumes that Postgres is running directly on the host machine? For those of us who run Postgres in Docker (Compose) it would be nice to have a script (or documentation including the script) that can work in that context.

I also run PG in docker, it should work unless you are using a different PG config. 
Assuming the standard I found for local dev, but can be modified to use .env

**Comment by Rotorsoft** [comment]
> This is awesome! Super cool idea elegantly implemented.
> 
> My only big-picture concern is that it's such an enormous ERD, it's difficult to view or navigate visually. Curious if you have any advice, or what you've been using to visually navigate the generated ERD.

I normally use the mermaid live editor - [https://mermaid.live](https://mermaid.live) , but the Github integration should be good enough. There is also the Miro and VSCode integrations... [https://mermaid.js.org/ecosystem/integrations-community.html](https://mermaid.js.org/ecosystem/integrations-community.html)

File: packages/commonwealth/scripts/gen-mermaid-erd.sh (hunk 3 above)
Comment (Rotorsoft) [reply packages/commonwealth/scripts/gen-mermaid-erd.sh:5]: Makes sense!

**Comment by rbennettcw** [comment]
> > Nice.
> > Looks like the script assumes that Postgres is running directly on the host machine? For those of us who run Postgres in Docker (Compose) it would be nice to have a script (or documentation including the script) that can work in that context.
> 
> I also run PG in docker, it should work unless you are using a different PG config. Assuming the standard I found for local dev, but can be modified to use .env

Do you have `psql` installed on the host?

File: packages/commonwealth/package.json (hunk 1 above)
Comment (Rotorsoft) [reply packages/commonwealth/package.json:19]: Sorry, missed that step!

**Comment by Rotorsoft** [comment]
> > > Nice.
> > > Looks like the script assumes that Postgres is running directly on the host machine? For those of us who run Postgres in Docker (Compose) it would be nice to have a script (or documentation including the script) that can work in that context.
> > 
> > 
> > I also run PG in docker, it should work unless you are using a different PG config. Assuming the standard I found for local dev, but can be modified to use .env
> 
> Do you have `psql` installed on the host?

yes

**Review by gdjohnson — APPROVED** [review]

**Comment by gdjohnson** [comment]
Looks like a minor merge conflict, otherwise looks good from a documentation perspective, thanks for the swift changes.

I wasn't having great luck with an ERD this size using Matt Bierner's Markdown Mermaid Support extension for VSCode, but I'll give the official Mermaid extension a shot. Either way, a minor consideration.

**Review by rbennettcw — APPROVED** [review]

**Pull request merged**